blockprod/src/detail/mod.rs

```diff
@@ -141,14 +143,22 @@ impl BlockProduction {
                         })
                     };
 
+                    let block_height = best_block_index.block_height().next_height();
+                    let sealed_epoch_randomness = this.get_epoch_data(block_height);
+                    // TODO this doesn't exist, thinking of adding to chainstate trait
```

Comment:
This is what I was referring to, about adding to the chainstate trait (read only)